java/core/src/main/java/sleeper/core/deploy/LambdaJar.java

```diff
@@ -51,9 +56,11 @@ public class LambdaJar {
     private final String filename;
     private final String imageName;
     private final boolean alwaysDockerDeploy;
+    private final String filenameFormat;
 
-    private LambdaJar(String filename, String imageName, boolean alwaysDockerDeploy) {
-        this.filename = Objects.requireNonNull(filename, "filename must not be null");
+    private LambdaJar(String filenameFormat, String version, String imageName, boolean alwaysDockerDeploy) {
+        this.filenameFormat = Objects.requireNonNull(filenameFormat, "filename must not be null");
+        this.filename = Objects.requireNonNull(String.format(filenameFormat, SleeperVersion.getVersion()), "filename must not be null");
```

Comment:
It doesn't look like there's a need to call requireNonNull here, that's checked in the line above.